server/handlers/user_handler.go

```diff
@@ -9,21 +9,24 @@ import (
 
 	"github.com/gorilla/mux"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 
 	"github.com/layer5io/meshery/server/models"
 )
 
 // UserHandler returns info about the logged in user
-func (h *Handler) UserHandler(w http.ResponseWriter, _ *http.Request, _ *models.Preference, user *models.User, _ models.Provider) {
```

Comment:
Why remove this?
In session middleware we are already fetching the user detail and passing, why can’t we use that??
https://github.com/meshery/meshery/blob/f7d0a28a250e7c9a078601e864aa68486b349ae2/server/handlers/middlewares.go#L232

Reply:
Firstly, it is not removed, and the reason I don't use the Session middleware is that it serves us a different use case.

Reply:
By removing, I meant why you are removing the caching logic?
Also, you have the user details (struct); you are just re-fetching, and I don't understand why.
func (h *Handler) UserHandler(w http.ResponseWriter, r *http.Request, _ *models.Preference, ------>_ *models.User, provider models.Provider)